synadm/cli/__init__.py

```diff
@@ -172,35 +161,30 @@ def root(ctx, verbose, output, config_file):
 @click.option('--admin-api-path', '-p', type=str, default='/_synapse/admin',
     help="""the path Synapse provides its admin API's, usually the default is
     alright for most installations.""", show_default=True)
-@click.option('--view', type=click.Choice(['table', 'raw']), default='raw',
+@click.option('--output', type=click.Choice(['human', 'json', 'yaml', 'pprint']), default='human',
     help="""how should synadm display data by default? 'table' gives a
     tabular view but needs your terminal to be quite width. 'raw' shows
     formatted json exactely as the API responded. Note that this can always
     be overridden by using global switches -r and -t (eg 'synadm -r user
     list')""", show_default=True)
 @click.pass_obj
-def config(api, user, token, base_url, admin_api_path, view):
+def config(api, user, token, base_url, admin_api_path, output):
     """modify synadm's configuration. configuration details are asked
     interactively but can also be provided using command line options."""
     click.echo('Running configurator...')
-    cfg = api.config
-    # get defaults for prompts from either config file or commandline
-    user_default = cfg.user if cfg.user else user
-    token_default = cfg.token if cfg.token else token
-    base_url_default = cfg.base_url if cfg.base_url else base_url
-    admin_path_default = cfg.admin_path if cfg.admin_path else admin_api_path
-    view_default = cfg.view if cfg.view else view
-    # prompts
-    api_user = click.prompt("Synapse admin user name", default=user_default)
-    api_token = click.prompt("Synapse admin user token", default=token_default)
-    api_base_url = click.prompt("Synapse base URL", default=base_url_default)
-    api_admin_path = click.prompt("Synapse admin API path", default=admin_path_default)
-    api_view = click.prompt("How should data be viewed by default?",
-           default=view_default, type=click.Choice(['table', 'raw']))
-
-    conf_dict = {"user": api_user, "token": api_token, "base_url": api_base_url,
-          "admin_api_path": api_admin_path, "view": api_view}
-    cfg.write(conf_dict)
+    api.write_config({
+        "user": click.prompt("Synapse admin user name",
+            default=api.config.get("user", user)),
+        "token": click.prompt("Synapse admin user token",
+            default=api.config.get("token", token)),
+        "base_url": click.prompt("Synapse base URL",
+            default=api.config.get("base_url", base_url)),
+        "admin_api_path": click.prompt("Synapse admin API path",
+            default=api.config.get("admin_api_path", admin_api_path)),
+        "format": click.prompt("How should data be viewed by default?",
+            default=api.config.get("format", output),
+            type=click.Choice(['human', 'json', 'yaml', 'pprint']))
+    })
```

Comment:
After writing new config to file it needs to be read again here otherwise if config was called automatically because a user issued any command with an incomplete /broken config file, after going through the configurator it would fail. This fixes it: `helper.config.load()'